src/mongo/MongoScrapbook.ts

```diff
@@ -372,23 +374,24 @@ class FindMongoCommandsVisitor extends MongoVisitor<MongoCommand[]> {
 			} catch (error) {
 				let err: IParsedError = parseError(error);
 				this.addErrorToCommand(err, ctx);
+				return {};
 			}
 		}
-		return constructedObject;
+		return { $oid: constructedObject.toString() };
 	}
 
 	private regexLiteralContextToObject(ctx: mongoParser.ArgumentContext | mongoParser.PropertyValueContext, text: string): Object {
 		let separator = text.lastIndexOf('/');
 		let flags = separator !== text.length - 1 ? text.substring(separator + 1) : "";
 		let pattern = text.substring(1, separator);
-		let tokenObject: Object = {};
 		try {
-			tokenObject = new RegExp(pattern, flags);
+			let tokenObject = new RegExp(pattern, flags);
+			return { $regex: tokenObject.source, $options: tokenObject.flags };
```

Comment:
Why tokenObject.flags instead of just flags?

Reply:
Here's my thought process. 

In your suggestion an invalid pattern ([example](https://github.com/Microsoft/vscode-cosmosdb/blob/master/test/mongoGetCommand.test.ts#L739)) would still be checked for by `EJSON.parse`. 

Using the Regexp constructor helps because: 
1. The errors thrown by Regexp would be closer to the root cause rather than wrapped by EJSON.
2. Throwing an error here adds to the problems view with the correct position, handling this at EJSON would not. 

This is a bit non-obvious - any suggestion on making this clearer?